geth/jail/cell.go

```diff
@@ -86,11 +86,37 @@ func (c *Cell) Stop() error {
 // event queue loop and schedules for immediate execution.
 // Intended to be used by any cell user that want's to run
 // async call, like callback.
-func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	task := looptask.NewCallTask(fn, args...)
-	// Add a task to the queue.
-	c.loop.Add(task)
-	// And run the task immediately.
-	// It's a blocking operation.
-	c.loop.Ready(task)
+	errChan := make(chan error, 1)
+
+	go func() {
+		err := c.loop.Add(task)
+		if err != nil {
+			errChan <- err
```

Comment:
Should we `return` after `errChan <- err`?

Reply:
Yes. Added return statement.